frontend/app-development/features/appPublish/components/Deploy/DeployDropdown/DeployDropdown.test.tsx

```diff
@@ -0,0 +1,207 @@
+    await user.click(select);
+
+    expect(screen.getByRole('option', { name: imageOptions[0].label })).toBeInTheDocument();
+    expect(screen.getByRole('option', { name: imageOptions[1].label })).toBeInTheDocument();
+  });
+
+  it('selects default image option', async () => {
+    render({ selectedImageTag: imageOptions[0].value });
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    expect(screen.getByRole('combobox')).toHaveValue(imageOptions[0].label);
+  });
+
+  it('selects new image option', async () => {
+    const user = userEvent.setup();
+
+    render();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    const select = screen.getByLabelText(textMock('app_deployment.choose_version'));
+    await user.click(select);
+
+    const option = screen.getByRole('option', { name: imageOptions[1].label });
+    await user.click(option);
+
+    await waitFor(() => {
+      expect(defaultProps.setSelectedImageTag).toHaveBeenCalledWith(imageOptions[1].value);
+    });
+  });
+
+  it('shows a loding spinner when mutation is pending', async () => {
+    render({ isPending: true });
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    const deployButton = screen.getByRole('button', {
+      name: textMock('app_deployment.btn_deploy_new_version'),
+    });
+    expect(within(deployButton).getByTestId('spinner-test-id')).toBeInTheDocument();
+  });
+
+  it('disables both dropdown and button when deploy is not possible', async () => {
+    render({ disabled: true });
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    expect(screen.getByLabelText(textMock('app_deployment.choose_version'))).toBeDisabled();
+
+    const deployButton = screen.getByRole('button', {
+      name: textMock('app_deployment.btn_deploy_new_version'),
+    });
+    expect(deployButton).toBeDisabled();
+  });
+
+  it('should confirm and close the dialog when clicking the confirm button', async () => {
+    const user = userEvent.setup();
+
+    render();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    const deployButton = screen.getByRole('button', {
+      name: textMock('app_deployment.btn_deploy_new_version'),
+    });
+    await user.click(deployButton);
+
+    const confirmButton = screen.getByRole('button', { name: textMock('general.yes') });
+    await user.click(confirmButton);
+
+    expect(defaultProps.startDeploy).toHaveBeenCalledTimes(1);
+    await waitFor(() => expect(screen.queryByRole('dialog')).not.toBeInTheDocument());
+  });
+});
+
+const render = (props?: Partial<DeployDropdownProps>, queries?: Partial<ServicesContextProps>) => {
```

Comment:
What about being more explicit with naming?
```suggestion
const renderDeployDropdown = (props?: Partial<DeployDropdownProps>, queries?: Partial<ServicesContextProps>) => {
```